packages/runtime/src/utils/partition.ts

```diff
@@ -0,0 +1,43 @@
+type Predicate<T, U extends T> = (value: T) => value is U
+
+export function partition<T, U extends T>(
+  array: readonly T[],
+  predicate: Predicate<T, U>,
+): [U[], Exclude<T, U>[]]
+
+export function partition<T>(array: readonly T[], predicate: (value: T) => boolean): [T[], T[]]
+
+export function partition(
```

Comment:
May be helpful to add a JSDoc string clarifying that the first returned array is the matches, and that the results are returned in stable order.

Reply:
Good point, we should probably document all of the general-purpose "utils" functions.

Reply:
This particular convention for `partition` is a de facto standard, but docs never hurt.